security-proxy/README.md

```diff
@@ -3,20 +3,77 @@
+
+### Proxy configuration
+
+It may sound obvious, but the new application has to be proxified by the security-proxy first !
+This can be configured in the [targets-mapping](https://github.com/georchestra/datadir/blob/master/security-proxy/targets-mapping.properties) file, which can be found in geOrchestra datadir. Remember: changes to this file require the security proxy to be restarted.
+This file maps public URLs to internal (private) URLs.
+For instance, the `header=http://localhost:8280/header/` line means "requests hitting the /header path should be routed to http://localhost:8280/header".
+
+Now, imagine your application has a public frontend (`/newapp/frontend`) and a private backend (`/newapp/backend`).
+You probably would like to restrict backend access to administrators, or people having a specific role.
+
+This can be done very easily with the [security-mappings](https://github.com/georchestra/datadir/blob/master/security-proxy/security-mappings.xml) file, also from the geOrchestra datadir.
+In this file, the last line starting with `<intercept-url pattern=".*" access="IS_AUTHENTICATED_ANONYMOUSLY,ROLE_USER...` means "by default, grant access to every path for anonymous and authenticated users". This means the frontend app is already available to every user, once proxified.
+
+To restrict access to the backend, one just has to insert a new line before the last one.
+Here we choose to restrict backend access to people having the `ADMINISTRATOR` role:
+```xml
+<intercept-url pattern="/newapp/backend" access="ROLE_ADMINISTRATOR" />
+```
+
+It is also possible to create a specific role which grants access to the backend, eg with role `NEWAPP_ADMIN`. The rule becomes:
+```xml
+<intercept-url pattern="/newapp/backend" access="ROLE_NEWAPP_ADMIN" />
+```
+
+### Application configuration
+
+#### Headers
+
+With every request, the proxy adds specific HTTP headers, so the application knows:
+ * if the request comes from a registered user, or an anonymous one - this is `sec-username` (not provided if anonymous).
+ * which roles the user bears - `sec-roles` is a semi-colon separated list of roles (not provided if anonymous).
+ * which organisation the user belongs to - `sec-orgname` provides the human-readable organisation title while `sec-org` is mapped onto the organisation id (LDAP's cn).
+
+Several other user properties are also provided as headers:
+ * `sec-email` is the user email
+ * `sec-firstname` is the first name (LDAP `givenName`)
+ * `sec-lastname` is the second name (LDAP `sn`)
+ * `sec-tel` is the user phone number (LDAP `telephoneNumber`)
+
+Additional headers can be configured in the proxy with the [headers-mappings](https://github.com/georchestra/datadir/blob/master/security-proxy/headers-mappings.xml) file.
+
+With the headers an application receives, it should be able to handle requests appropriately.
```

Comment:
The application handles the requests appropriatly thanks to the headers received.